CONTRIBUTING_GUIDELINES.md

```diff
@@ -147,3 +150,8 @@ git push --force
 
 [good-commit-msg]: https://chris.beams.io/posts/git-commit/
 [dco-bot]: https://github.com/probot/dco#how-it-works
+
+## Branches
+
+Upstream repository should contain only maintenance branches (e.g. `release-1.0`). For feature
+branches use forked repository.
```

Comment:
We may want to add a bit of explanation here. In particular, one critical step that does not happen automatically with a fork is to make sure the local `master` branch tracks the upstream, not your fork (origin).
```shell
$ git remote -v
origin	git@github.com:jaegertracing/jaeger.git (fetch)
origin	git@github.com:jaegertracing/jaeger.git (push)

$ git remote rm origin

$ git remote add upstream git@github.com:jaegertracing/jaeger.git

$ git remote add origin git@github.com:yurishkuro/jaeger.git

$ git fetch upstream master
remote: Enumerating objects: 13, done.
remote: Counting objects: 100% (13/13), done.
remote: Total 19 (delta 12), reused 13 (delta 12), pack-reused 6
Unpacking objects: 100% (19/19), done.
From github.com:jaegertracing/jaeger
 * branch            master     -> FETCH_HEAD
 * [new branch]      master     -> upstream/master

$ git pull
There is no tracking information for the current branch.
Please specify which branch you want to merge with.
See git-pull(1) for details.

    git pull <remote> <branch>

If you wish to set tracking information for this branch you can do so with:

    git branch --set-upstream-to=<remote>/<branch> master

$ git branch --set-upstream-to=upstream/master master
Branch 'master' set up to track remote branch 'master' from 'upstream'.

$ git remote -v
origin    git@github.com:yurishkuro/jaeger.git (fetch)
origin    git@github.com:yurishkuro/jaeger.git (push)
upstream  git@github.com:jaegertracing/jaeger.git (fetch)
upstream  git@github.com:jaegertracing/jaeger.git (push)
```

Without this, the local master tracks for fork and you need to do extra steps to merge upstream master into it. People often forget to do that and end up with PRs having all kinds of weird commits in them.